### ISSUE TITLE
Missing easy way to set various renderers/samplers

### ISSUE THREAD

**Issue by paciox** [huggingface/diffusers#283]
### Describe the bug

(https://www.reddit.com/r/StableDiffusion/comments/wwm2at/sampler_vs_steps_comparison_low_to_mid_step_counts/?utm_medium=android_app&utm_source=share)

Check here, now

```
Now I launch it like this:
import sys
import torch
from torch import autocast
from diffusers import StableDiffusionPipeline

model_id = "CompVis/stable-diffusion-v1-4"
device = "cuda"


pipe = StableDiffusionPipeline.from_pretrained(model_id, use_auth_token=True)
pipe = pipe.to(device)

image1 = pipe("SOME PROPMT", num_inference_steps=50, width=512, height=512)["sample"][0] 
  image1.save(f"aa.png")

```

NOw what if I want to set the others?

### System Info

```shell
Windows 10, SD with 1.4 weights
```

**Comment by anton-l** [comment]
Hi @paciox! There are a couple of ways to set a custom/alternative sampler for a pipeline:

1. Initialize another scheduler from scratch and pass it to the pipeline loader:
```python
from diffusers import StableDiffusionPipeline, LMSDiscreteScheduler

model_id = "CompVis/stable-diffusion-v1-4"
# Use the K-LMS scheduler here instead
scheduler = LMSDiscreteScheduler(beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear", num_train_timesteps=1000)
pipe = StableDiffusionPipeline.from_pretrained(model_id, scheduler=scheduler, use_auth_token=True)
pipe = pipe.to("cuda")
```

2. Load the scheduler's hyperparameters from another scheduler's config. Here we're loading the default scheduler's (PNDM's) params like `beta_start, beta_end` into LMS:
```python
from diffusers import StableDiffusionPipeline, LMSDiscreteScheduler

model_id = "CompVis/stable-diffusion-v1-4"
scheduler = LMSDiscreteScheduler(model_id, subfolder="scheduler", use_auth_token=True)
pipe = StableDiffusionPipeline.from_pretrained(model_id, scheduler=scheduler, use_auth_token=True)
pipe = pipe.to("cuda")
```
The second only works for compatible schedulers, e.g. the ones that use beta schedules in this example.

**Comment by patrickvonplaten** [comment]
This is also related to: https://github.com/huggingface/diffusers/issues/336
